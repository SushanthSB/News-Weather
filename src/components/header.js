import React from 'react';
import '../styles/header.css'

//Header part component that displays brand name

function handleOpen() {
	document.querySelector(".side-nav").style.width="200px";
}
export default (props) => {
	return(
			<nav ref={props.navRef} className='navbar navbar-default navbar-fixed-top header-bar'>
						<h1  ref={props.headerBrandRef} className='header-brand'><span>Daily News</span>
						<span>
							<button type="button" className="navbar-toggle" onClick={handleOpen} >
      	  						<div className="icon-bar"></div>
        						<div className="icon-bar"></div>
        						<div className="icon-bar"></div>                        
      						</button>
						</span>
						</h1>
			</nav>
		)
}