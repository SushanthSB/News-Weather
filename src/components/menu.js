import React from 'react';
import PropTypes from 'prop-types'

// Menu component - displays menu list(General, Business, Technology etc...)
class Menu extends React.Component {

	handleClick(term) {
		//call dataFetch action via parent component
		this.props.handleTerm(term,this.props.subCategory[0][term][0])
	}
	
	category() {
		return (
				this.props.category.map((name, index) => {
					//make className active if clicked item and diplayed item in the corresponding list are same
					return (<li key={index} className={(this.props.menu.term === name.name) ? 'active':'not-active'} onClick={this.handleClick.bind(this, name.name)} ><a>{name.name}</a></li>)
				})
			)
	}
	
	componentDidMount() {
		// call dataFetch aciton with default value 'General'
		this.props.handleTerm('General',this.props.subCategory[0]['General'][0]);
	}

	render() {
		return (
				<ul className='nav nav-tabs'>{this.category()}</ul>
			)
	}
}

Menu.propTypes = {
  category: PropTypes.array.isRequired,
  subCategory: PropTypes.array.isRequired,
  menu: PropTypes.object.isRequired,
} 

export default Menu