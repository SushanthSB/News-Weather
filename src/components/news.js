import React from 'react';
import '../styles/news.css';
import PropTypes from 'prop-types';
 
 //News component- Fetches and displays news.
const News = (props) => {

	const news = () => {
		return (
				props.fetch.data.map((news, index) => {
					return (
						<div className='panel panel-default news-default col-xs-12 col-sm-12 col-md-12 col-lg-12' key={index}>
							<a href={news.url} className='col-xs-12 col-sm-12 col-md-12 col-lg-12' >
								<div className='news-img col-xs-2 col-sm-2 col-md-2 col-lg-2' >
									<img src={news.urlToImage} alt={news.urlToImage} />
								</div>
								<div className='col-xs-9 col-xs-offset-1 col-sm-9 col-md-9 col-lg-9' >
									<div className='panel-heading news-title' > {news.title} </div>
									<div className='panel-body news-body' > {news.description} </div>
								</div>
							</a>
						</div>
					)
				})
			)
		}

	return(
			<div className='panel-group news-group col-xs-12 col-sm-12 col-md-8 col-md-offset-1 col-lg-8 col-lg-offset-1' >
				<h3 className='top-stories'>Top Stories</h3>
				{news()}
			</div>
		)
}

News.propTypes = {
	fetch: PropTypes.object.isRequired
}

export default News;