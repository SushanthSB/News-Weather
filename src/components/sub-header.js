import React from 'react';
import '../styles/sub-header.css';
import Typing from 'react-typing-animation';
import PropTypes from 'prop-types';

//sub-header content - Displays animated typing
const AnimatedTypingComponent = (sources) => {
  return(
    sources.map((source, index) => {
       return(
          <span  key={index} >
            <span>{source}</span>
            <Typing.Backspace count={50} delay={2000} speed={30} />
            <Typing.Delay ms={0} />
            <Typing.Speed ms={130} />
          </span>
         )
     })
    ) 
}

const SubHeader = (props) => {
  return (
          <div className='jumbotron sub-header col-xs-12 col-sm-12 col-md-12 col-lg-12'>
              <p> Get live headlines from </p>
              {/*Fetch values from news-sourcce reducer and pass it to react-typing-animation library*/}
              <h1>  
                <Typing loop={true} speed={130} startDelay={-10} >
                  {AnimatedTypingComponent(props.news_source)}
                </Typing >
              </h1>
          </div>
      )
}

SubHeader.propTypes = {
  news_source: PropTypes.array.isRequired
}

export default SubHeader