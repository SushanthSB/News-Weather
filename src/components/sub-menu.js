import React from 'react';
import '../styles/sub-menu.css'
import PropTypes from 'prop-types'

//submenu component - Displays submenu(The Hindu, Abc-news-au etc...)
class SubMenu extends React.Component {
	handleClick(term,subterm) {
		this.props.handleTerm(term,subterm)
	}
	subCategory() {
		var arr=this.props.subCategory[0][this.props.menu.term];
		if(arr) {
			return (
				arr.map((submenu,index) => {
					return <li key={index} 
					//activate class name if the clicked component and the rendered li component are equal
					className={(this.props.menu.subterm === submenu)?'active':'in-active'} 
					onClick={this.handleClick.bind(this, this.props.menu.term, submenu)} >
					<a>{submenu}</a>
					</li>
				})
			)
		}
	}
	render() {
		return (
				<ul className='nav nav-pills submenu' >{this.subCategory()}</ul>
			)
	}
}

SubMenu.propTypes = {
  subCategory: PropTypes.array.isRequired,
  menu: PropTypes.object.isRequired,
  handleTerm: PropTypes.func
}

export default SubMenu;