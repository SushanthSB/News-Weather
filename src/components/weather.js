import React from 'react';
import '../styles/weather.css';
import PropTypes from 'prop-types';

//weather component that fetches and diplays weather 
export const celToFar = (val) => {
	return ((val*1.8+32).toFixed(2))
}

export const today = () => {
	const days=['sunday','monday','tuesday','wednesday','thursday','friday','saturday'];
	return days[new Date().getDay()].toUpperCase();
}

export const Weather = (props) => {
		return(
				<div className=''>
				<h3 className='weather-brand'>Weather</h3>
				<div className='jumbotron weather-div1 col-md-3 col-lg-3' >
					<div className='city-name col-xs-12 col-sm-12 col-md-12 col-lg-12'>
						<h3>{props.weather.data.long_name}</h3>
					</div>
					<div className='weather-div2 col-xs-12 col-sm-12 col-md-12 col-lg-12' >
						<table className='col-xs-12 col-sm-12 col-md-12 col-lg-12 weather-table' >
							<thead>
								<tr>
									<th colSpan='2'>{today()}</th>
								</tr>
							</thead>
							<tbody>
								<tr className='temp'>
									<td colSpan='2' className='temp-img'><img src={'http://openweathermap.org/img/w/'+props.weather.weather.icon+'.png'} alt={props.weather.weather.description} />
									<span>{Math.floor(props.weather.temp.temp)}</span></td>
								</tr>
								<tr className='temp-desc' >
									<td colSpan='2'>{props.weather.weather.description}</td> 
								</tr>
								<tr className='temp-rem' >
									<td>Temp:</td> 
									<td>{props.weather.temp.temp}&#8451; | {celToFar(props.weather.temp.temp)}&#8457;</td>
								</tr>
								<tr className='temp-rem'>
									<td>Min|Max:</td> 
									<td>{props.weather.temp.temp_min}&#8451; | {props.weather.temp.temp_max}&#8451;</td>
								</tr>
								<tr className='temp-rem'>
									<td>Min|Max:</td>
									<td>{celToFar(props.weather.temp.temp_min)}&#8457; | {celToFar(props.weather.temp.temp_max)}&#8457;</td>
								</tr>
								<tr className='temp-rem'>
									<td>Humidity</td>
									<td>{props.weather.temp.humidity}%</td>
								</tr>
							</tbody>
						</table>
					</div>
				</div>
				</div>
			)
}

Weather.propTypes = {
	weather: PropTypes.object.isRequired
}