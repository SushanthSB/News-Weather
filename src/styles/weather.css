.weather-div1 {
	padding:10px 20px 30px 10px !important;
	background-color: white;
	border: 1px solid #D6D6D6;
}
.city-name h3 {
	text-align: center;
	font-size: 40px;
}

.weather-brand {
	margin-top: 25px;
	font-size:30px;
}

.weather-div2 {
	margin-top: 20px;
}

.weather-table th {
	text-align: center;
}
.weather-table .temp-rem td {
	text-align: center;
	font-size: 20px;
}
.weather-table .temp .temp-img {
	text-align: center;
}
.weather-table .temp .temp-img img {
	width:90px;
	vertical-align: text-bottom;
}
.weather-table .temp .temp-img span {
	font-size: 90px;
}
.weather-table .temp-rem td {
	padding:10px 0px;
	font-size:20px;
}
.weather-table .temp-desc td {
	font-size:25px;
	text-align: center
}
.weather-table thead tr th {
	font-size: 25px;
	font-weight: normal;
	letter-spacing: 5px;
}