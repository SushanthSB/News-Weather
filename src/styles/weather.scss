.weather-brand {
	margin-top: 25px;
	font-size:30px;
}

.weather-div1 {
	padding:10px 20px 30px 10px !important;
	background-color: white;
	border: 1px solid #D6D6D6;
	.city-name h3 {
		text-align: center;
		font-size: 40px;
	}
	.weather-div2 {
		margin-top: 20px;
		.weather-table th {
			text-align: center;
			font-size: 25px;
			font-weight: normal;
			letter-spacing: 5px;
		}
		.weather-table .temp {
			.temp-img {
				text-align: center;
				img {
					width:100px;
					vertical-align: text-bottom;
				}
				span {
					font-size: 90px;
				}
			}
		}
		.weather-table .temp-desc td {
			font-size:25px;
			text-align: center
		}	
		.weather-table .temp-rem td {
			text-align: center;
			font-size: 20px;
			padding:10px 0px;
			font-size:20px;
		}
	}
}




