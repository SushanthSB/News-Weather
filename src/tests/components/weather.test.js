import React from 'react';
import { shallow } from 'enzyme';
import { Weather, celToFar, today } from '../../components/weather';
import renderer from 'react-test-renderer';
import { spy } from 'sinon'

describe("Weather",() => {
	let props = {
		weather:{
       		"data":{"log_name":"Bengaluru"},
       		"weather":{"description":"sunny","icon":"04n"},
       		"temp":{"temp":22.45,"temp_max":23,"temp_min":22}
       	}
	}
	it('renders without fail',() => {
		const tree = renderer.create(<Weather {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	})
	it('should call celToFar to convert celsius to farenheit on render',() => {
		let sp = spy(celToFar)
		sp(5)
		expect(sp.calledWith(5)).toEqual(true)
	})
	it('celToFar should take celsius and return farenheit of it',() => {
		const val = 5;
		const expected = (val*1.8+32).toFixed(2)
		expect(celToFar(val)).toEqual(expected)
	})
	it('should call today to fetch current day on render',() => {
		let sp = spy(today)
		sp();
		expect(sp.calledOnce).toEqual(true);
	})
	it('today should return current day',() => {
		const days=['sunday','monday','tuesday','wednesday','thursday','friday','saturday'];
		const expected = days[new Date().getDay()].toUpperCase();
		expect(today()).toEqual(expected)
	})
})
