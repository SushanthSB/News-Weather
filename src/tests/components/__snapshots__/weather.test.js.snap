// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather renders without fail 1`] = `
<div
  className="sim"
>
  <h3
    className="weather-brand"
  >
    Weather
  </h3>
  <div
    className="jumbotron weather-div1 col-md-3 col-lg-3"
  >
    <div
      className="city-name col-xs-12 col-sm-12 col-md-12 col-lg-12"
    >
      <h3 />
    </div>
    <div
      className="weather-div2 col-xs-12 col-sm-12 col-md-12 col-lg-12"
    >
      <table
        className="col-xs-12 col-sm-12 col-md-12 col-lg-12 weather-table"
      >
        <thead>
          <tr>
            <th
              colSpan="2"
            >
              SUNDAY
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            className="temp"
          >
            <td
              className="temp-img"
              colSpan="2"
            >
              <img
                alt="sunny"
                src="http://openweathermap.org/img/w/04n.png"
              />
              <span>
                22
              </span>
            </td>
          </tr>
          <tr
            className="temp-desc"
          >
            <td
              colSpan="2"
            >
              sunny
            </td>
          </tr>
          <tr
            className="temp-rem"
          >
            <td>
              Temp:
            </td>
            <td>
              22.45
              ℃ | 
              72.41
              ℉
            </td>
          </tr>
          <tr
            className="temp-rem"
          >
            <td>
              Min|Max:
            </td>
            <td>
              22
              ℃ | 
              23
              ℃
            </td>
          </tr>
          <tr
            className="temp-rem"
          >
            <td>
              Min|Max:
            </td>
            <td>
              71.60
              ℉ | 
              73.40
              ℉
            </td>
          </tr>
          <tr
            className="temp-rem"
          >
            <td>
              Humidity
            </td>
            <td>
              %
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
